firmware/src/maze/FlashIO.cc

```diff
@@ -0,0 +1,54 @@
+/*********************************************************************
+ * File:        FlashIO.cc
+ * Author:      Bryant Gonzaga
+ * Created:     3/23/2018
+ * Modified:    3/23/2018
+ *
+ * Notes:
+ *
+ * Description:
+ *  
+ ********************************************************************/
+
+#include "FlashIO.hpp"
+
+FlashIO::FlashIO()
+{
+    /* Initialize Variables */
+    dataStatus = NO_FLASH_DATA;
```

Comment:
We need something more than this. Imagine we write something in flash and power off, then on the next bootup, `dataStatus` should detect if the data is written last time and set `NEW_FLASH_DATA` instead of `NO_FLASH_DATA`. We can detect that by using a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)). I will show how it is done.